Tests/SonarQube.MSBuild.Tasks.UnitTests/MergeRuleSetsTests.cs

```diff
@@ -34,58 +37,77 @@ public void MergeRulesets_PrimaryRulesetDoesNotExist()
         }
 
         [TestMethod]
+        public void MergeRulesets_MergedRulesetAlreadyExists()
+        {
+            // Arrange
+            string projectDir = this.TestContext.TestDeploymentDir;
+            string primaryRuleset = this.CreateValidRuleset("valid.ruleset.txt");
+            string targetRulesetFilePath = Path.Combine(projectDir, "merged.ruleset.txt");
+            File.WriteAllText(targetRulesetFilePath, "dummy existing ruleset");
+
+            DummyBuildEngine dummyEngine = new DummyBuildEngine();
+            MergeRuleSets task = CreateTask(dummyEngine, projectDir, primaryRuleset, targetRulesetFilePath);
+
+            // Act and Assert
+            InvalidOperationException ex = AssertException.Expects<InvalidOperationException>(() => task.Execute());
+            Assert.IsTrue(ex.Message.Contains(targetRulesetFilePath));
+        }
```

Comment:
This is the only new test in the file.
The other changes are to take account of the fact that the task now expects the name of the merged file to be supplied, and requires it to be different from the *primaryRuleset*.